src/overlays/actors/ovl_Obj_Bean/z_obj_bean.c

```diff
@@ -16,13 +17,73 @@ void ObjBean_Destroy(Actor* thisx, GlobalContext* globalCtx);
 void ObjBean_Draw(Actor* thisx, GlobalContext* globalCtx);
 
-extern UNK_TYPE D_06000090;
-extern UNK_TYPE D_060001B0;
-extern UNK_TYPE D_060003F0;
-extern UNK_TYPE D_060005DC;
-extern UNK_TYPE D_06000650;
+void ObjBean_WaitForPlayer(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_Fly(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupFly(ObjBean* this);
+void ObjBean_WaitForWater(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupWaitForWater(ObjBean* this);
+void ObjBean_SetupGrowWaterPhase1(ObjBean* this);
+void ObjBean_GrowWaterPhase1(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_GrowWaterPhase2(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupGrowWaterPhase2(ObjBean* this);
+void ObjBean_GrowWaterPhase3(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupGrowWaterPhase3(ObjBean* this);
+void ObjBean_SetupGrown(ObjBean* this);
+
+void ObjBean_FlattenLeaves(ObjBean* this);
+void ObjBean_Grown(ObjBean* this);
+void ObjBean_LeavesStill(ObjBean* this);
+void ObjBean_SetupShakeLeaves(ObjBean* this);
+void ObjBean_ShakeLeaves(ObjBean* this);
+void ObjBean_SetupWaitForBean(ObjBean* this);
+void ObjBean_WaitForBean(ObjBean* this, GlobalContext* globalCtx);
+void func_80B8FE3C(ObjBean* this, GlobalContext* globalCtx);
+void func_80B8FE00(ObjBean* this);
+void func_80B8FE6C(ObjBean* this);
+void func_80B8FEAC(ObjBean* this, GlobalContext* globalCtx);
+void func_80B8FF50(ObjBean* this);
+void ObjBean_SetupGrowWaterPhase4(ObjBean* this);
+void func_80B8FF8C(ObjBean* this, GlobalContext* globalCtx);
+void func_80B90050(ObjBean* this, GlobalContext* globalCtx);
+void func_80B90010(ObjBean* this);
+void func_80B908EC(ObjBean* this);
+void func_80B90918(ObjBean* this, GlobalContext* globalCtx);
+void func_80B90970(ObjBean* this);
+void func_80B909B0(ObjBean* this, GlobalContext* globalCtx);
+void func_80B909F8(ObjBean* this);
+void func_80B90A34(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupWaitForPlayer(ObjBean* this);
+void ObjBean_GrowWaterPhase4(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_GrowWaterPhase5(ObjBean* this, GlobalContext* globalCtx);
+void ObjBean_SetupGrowWaterPhase5(ObjBean* this);
+void ObjBean_SetupShakeLeavesFast(ObjBean* this);
+void ObjBean_ShakeLeavesFast(ObjBean* this);
+void ObjBean_Grow(ObjBean* this);
+void ObjBean_SetupGrow(ObjBean* this);
+void ObjBean_SetupWaitForStepOff(ObjBean* this);
+void ObjBean_WaitForStepOff(ObjBean* this, GlobalContext* globalCtx);
+
+extern Gfx D_06000090[];
+extern Gfx D_060001B0[];
+extern Gfx D_060003F0[];
+extern CollisionHeader D_060005DC;
+extern Gfx D_06000650[];
+
+
+#define BEAN_STATE_DRAW_LEAVES (1 << 0)
+#define BEAN_STATE_DRAW_SOIL (1 << 1)
+#define BEAN_STATE_DRAW_PLANT (1 << 2)
+#define BEAN_STATE_DRAW_STALK (1 << 3)
+
+#define BEAN_STATE_COLLIDER_SET (1 << 4)
+#define BEAN_STATE_DYNAPOLY_SET (1 << 5)
+
+#define BEAN_STATE_BEEN_WATERED (1 << 6)
+
+#define BEAN_STATE_PLAYER_ON_TOP (1 << 7)
```

Comment:
these all seem like the same thing and can be grouped together, id remove the breaks